daemon/daemon.go

```diff
@@ -283,6 +288,18 @@ func (daemon *Daemon) restore() error {
 	}
 	group.Wait()
 
+	removeGroup := sync.WaitGroup{}
+	for id := range removeContainers {
+		removeGroup.Add(1)
+		go func(cid string) {
+			defer removeGroup.Done()
```

Comment:
No need for `defer`, can just move after `ContainerRm`... probably for most people it would never even be noticeable in this case, but there is overhead with defer, so `for  ... { defer ... }` is generally not great.